pkg/operator/controller/controller.go

```diff
@@ -312,21 +310,35 @@ func (r *reconciler) enforceIngressFinalizer(ingress *operatorv1.IngressControll
 
 // ensureIngressDeleted tries to delete ingress, and if successful, will remove
 // the finalizer.
-func (r *reconciler) ensureIngressDeleted(ingress *operatorv1.IngressController, dnsConfig *configv1.DNS, infraConfig *configv1.Infrastructure) error {
-	if err := r.finalizeLoadBalancerService(ingress, dnsConfig); err != nil {
+func (r *reconciler) ensureIngressDeleted(ingress *operatorv1.IngressController) error {
+	if err := r.finalizeLoadBalancerService(ingress); err != nil {
 		return fmt.Errorf("failed to finalize load balancer service for %s: %v", ingress.Name, err)
 	}
 	log.Info("finalized load balancer service for ingress", "namespace", ingress.Namespace, "name", ingress.Name)
 
+	// Delete the wildcard DNS record, and block ingresscontroller finalization
+	// until the dnsrecord has been finalized.
+	if err := r.deleteWildcardDNSRecord(ingress); err != nil {
+		return fmt.Errorf("failed to delete wildcard dnsrecord: %v", err)
+	}
+	if record, err := r.currentWilcardDNSRecord(ingress); err != nil {
+		return fmt.Errorf("failed to get current wildcard dnsrecord: %v", err)
+	} else {
+		if record != nil {
+			log.V(1).Info("waiting for wildcard dnsrecord to be deleted", "dnsrecord", record.Name)
```

Comment:
Elsewhere we pass `record` in its entirety to `log.Info`.

Reply:
Fixed